packages/smooth_app/lib/pages/product/nutrition_page/nutrition_validator.dart

```diff
@@ -0,0 +1,482 @@
+
+  bool _validateSalt(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.G, s), 6.0);
+
+  bool _validateSaturatedFat(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.G, s), 20.0);
+
+  bool _validateSelenium(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MICRO_G, s), 400.0);
+
+  bool _validateSodium(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.G, s), 2.4);
+
+  bool _validateStearicAcid(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.G, s), 5.0);
+
+  bool _validateSugarAlcohol(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.G, s), 50.0);
+
+  bool _validateSugars(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.G, s), 90.0);
+
+  bool _validateTransFat(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.G, s), 2.0);
+
+  bool _validateVitaminA(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MICRO_G, s), 3000.0);
+
+  bool _validateVitaminB1(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MILLI_G, s), 50.0);
+
+  bool _validateVitaminB12(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MICRO_G, s), 2000.0);
+
+  bool _validateVitaminB2(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MILLI_G, s), 50.0);
+
+  bool _validateVitaminB6(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MILLI_G, s), 100.0);
+
+  bool _validateVitaminB9(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MICRO_G, s), 1000.0);
+
+  bool _validateVitaminC(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MILLI_G, s), 2000.0);
+
+  bool _validateVitaminD(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MICRO_G, s), 100.0);
+
+  bool _validateVitaminE(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MILLI_G, s), 1000.0);
+
+  bool _validateVitaminK(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MICRO_G, s), 120.0);
+
+  bool _validateVitaminPP(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MILLI_G, s), 35.0);
+
+  bool _validateZinc(double q, Unit u, double s) =>
+      _validateMax(_normalize(q, u, Unit.MILLI_G, s), 40.0);
+
+  double _normalize(
+      double quantity, Unit inputUnit, Unit typicalUnit, double servingWeight) {
+    final double factor = _unitConversionFactor(inputUnit, typicalUnit);
+    final double normalized = (quantity * factor) * (100.0 / servingWeight);
+    return normalized;
+  }
+
+  bool _validateMax(double normalizedValue, double maxAllowed) {
+    return normalizedValue <= maxAllowed;
+  }
+
+  double _unitConversionFactor(Unit from, Unit to) {
+    if (from == to) {
+      return 1.0;
+    }
+    // Mass conversions
+    if (from == Unit.G && to == Unit.MILLI_G) {
```

Comment:
I'm sure there are conversion methods somewhere else in the app.
Please use them.